tests/apps/test_data.py

```diff
@@ -209,3 +211,126 @@ def test_wp_non_negative(self, dataset):
+        [1, 1, 1, 1],
+        [2, 2],
+    ]
+
+    patch_feature_vectors = np.array(
+        [
+            [0.1, 0.01, 0, 0.1, 0.05],
+            [0.2, 0.02, 0, 0.05, 0.01],
+            [0.3, 0.05, 0, 0.05, 0.15],
+            [0.4, 0.15, 0, 0.15, 0.15],
+        ]
+    )
+
+    patch_adjacency_matrices = np.array(
+        [
+            [
+                [1.40801884, 1.37064318, 0.74647764, 0.8656113, 0.79760222],
+                [1.37064318, 1.14609425, 1.31463115, 0.48630012, 1.44034118],
+                [0.74647764, 1.31463115, 1.03330415, 1.64344102, 1.31083479],
+                [0.8656113, 0.48630012, 1.64344102, 0.28176324, 1.43776473],
+                [0.79760222, 1.44034118, 1.31083479, 1.43776473, 0.1894344],
+            ],
+            [
+                [0.07507458, 0.99955953, 1.1432025, 0.63240514],
+                [0.99955953, 1.71093457, 1.10163201, 0.99179716],
+                [1.1432025, 1.10163201, 0.07714951, 0.74115172],
+                [0.63240514, 0.99179716, 0.74115172, 0.41063534],
+            ],
+        ]
+    )
+
+    @pytest.fixture
+    def dataset_patched(self, monkeypatch, datasets):
+        """Fixture for loading each of the datasets in ``FEATURE_DATASETS_LIST`` with the
+        ``__init__`` monkeypatched to load the above ``patch_feature_vectors``."""
+
+        def mock_init(_self):
+            """Replacement ``__init__`` for all the datasets in ``FEATURE_DATASETS_LIST``"""
+            # pylint: disable=protected-access
+            _self.featuresData = self.patch_feature_vectors
+            _self.matData = self.patch_adjacency_matrices
+            _self.n_vectors, _self.n_features = 4, 5
+
+        with monkeypatch.context() as m:
+            m.setattr(datasets, "__init__", mock_init)
+            m.setattr(datasets, "unitData", self.patch_orbits)
+            yield datasets()
+
+    def test_data_filename(self, dataset):
+        """Test if file name is valid string for each dataset"""
+        # pylint: disable=protected-access
+        assert isinstance(dataset._data_filename, str)
+
+    def test_n_mean(self, dataset):
+        """Test if mean photon number is valid float or int for each dataset"""
+        assert isinstance(dataset.n_mean, (float, int))
+        assert dataset.n_mean >= 0
+
+    def test_threshold(self, dataset):
+        """Test if threshold flag is valid bool for each dataset"""
+        assert isinstance(dataset.threshold, bool)
+
+    def test_unit(self, dataset):
+        """Test if unit is valid string for each dataset"""
+        # pylint: disable=protected-access
+        assert isinstance(dataset.unit, str)
+        allowed = ["Orbits", "orbits", "Events", "events"]
+        assert dataset.unit in allowed
+
+    def test_method(self, dataset):
+        """Test if method is valid string for each dataset"""
+        # pylint: disable=protected-access
+        assert isinstance(dataset.method, str)
+        allowed = ["Exact", "exact", "MC", "mc"]

```

Comment:
I'm less keen on allowing both upper and lowercase and feel it should be enforced as one or the other (preferably lowercase).

The method is used for loading the file, so I think it's better to have a standard system.